spec/requests/user_request_spec.rb

```diff
@@ -19,4 +19,22 @@
     expect(response[:data]).to be_a(Hash)
     expect(response[:data][:users][0][:id].to_i).to be_a(Integer)
   end
+
+  it 'handles variables' do
+    post '/graphql', params: { query: '{ users {id, username } }', variables: 1000 }
+
+    response = JSON.parse(@response.body, symbolize_names: true)
+
+    expect(response[:data]).to be_a(Hash)
+    expect(response[:data][:users][0][:id].to_i).to be_a(Integer)
+  end
+
+  it 'handles variables' do
+    post '/graphql', params: { query: '{ users {id, username } }', variables: { var: 1000 } }
```

Comment:
this hits the variables:hash part of graphql controller

Reply:
Also no idea how this works but we've got it covered!

Reply:
![](https://media3.giphy.com/media/aMIImk0BKdik8/200.gif?cid=5a38a5a20gan0ced68r1tyahu34lliaslsxc6sxxlm6tk3u1&rid=200.gif)